### PR TITLE
Add PHP linting foundation

### PR DESCRIPTION
**Pull request by WyriHaximus**
Todo:
- [x] Linting PHP files
- [x] Tests

### CODE REVIEW

File: lib/linter.sh [hunk 1]
```diff
@@ -1298,6 +1322,18 @@ BuildFileList()
       ##########################################################
       READ_ONLY_CHANGE_FLAG=1
     ######################
+    # Get the PHP files #
+    ######################
+    elif [ "$FILE_TYPE" == "pl" ]; then
```
Comment (admiralAwkbar) [inline lib/linter.sh:1327]: Should this be `.php`?

Comment (WyriHaximus) [reply lib/linter.sh:1327]: Among things, there are also some other things that need to be changed. Currently working on getting the linter installed before moving on this fixing this file :smile:

File: Dockerfile [hunk 2]
```diff
@@ -92,6 +93,15 @@ RUN wget -qO- "https://github.com/koalaman/shellcheck/releases/download/stable/s
 ARG GO_VERSION='v1.23.7'
 RUN wget -O- -nvq https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s "$GO_VERSION"
 
+##################################
+# Install COmposer && PHP Linter #
```
Comment (staabm) [inline Dockerfile:97]:
```suggestion
# Install Composer && PHP Linter #
```

File: lib/linter.sh [hunk 3]
```diff
@@ -2098,14 +2136,14 @@ if [ "$VALIDATE_PYTHON" == "true" ]; then
 fi
 
 ################
-# PERL LINTING #
```
Comment (staabm) [inline lib/linter.sh:2101]: you should have copied this part before editing it :)

Comment (WyriHaximus) [reply lib/linter.sh:2101]: image

**Comment by brzuchal** [comment]
Wouldn't it be cleaner if the `php-parallel-lint/php-parallel-lint` got improved with phar building and getting that from GH releases? Then it won't need a Composer nor `php7-curl`, `php7-openssl` and `php7-zip`!

**Comment by WyriHaximus** [comment]
> Wouldn't it be cleaner if the `php-parallel-lint/php-parallel-lint` got improved with phar building and getting that from GH releases? Then it won't need a Composer nor `php7-curl`, `php7-openssl` and `php7-zip`!

Well, actually it turns out using a parallel linter is no use at all since the super-linter runs the linters for a single file at a time. So fell back to the buildin `-l`. As for installing composer and the other extra packages, I'm not to worry about about adding a few extra seconds to the build that's already clocking in at 5 minutes. But it would indeed be cleaner (and faster) to use a phar if it's available.

File: README.md [hunk 4]
```diff
@@ -16,23 +16,24 @@ The super-linter finds issues and reports them to the console output. Fixes are
...
+| ---              | ---                                                                         |
+| **Ansible**      | [ansible-lint](https://github.com/ansible/ansible-lint)                     |
+| **CoffeeScript** | [coffeelint](http://www.coffeelint.org/)                                    |
+| **Dockerfile**   | [dockerfilelint](https://github.com/replicatedhq/dockerfilelint.git)        |
+| **Golang**       | [golangci-lint](https://github.com/golangci/golangci-lint)                  |
+| **JavaScript**   | [eslint](https://eslint.org/) [standard js](https://standardjs.com/)        |
+| **JSON**         | [jsonlint](https://github.com/zaach/jsonlint)                               |
+| **Markdown**     | [markdownlint](https://github.com/igorshubovych/markdownlint-cli#readme)    |
+| **Perl**         | [perl](https://pkgs.alpinelinux.org/package/edge/main/x86/perl)             |
+| **PHP**          | [PHP-Parallel-Lint](https://github.com/php-parallel-lint/PHP-Parallel-Lint) |
```
Comment (alfredbez) [inline README.md:29]: Looks like you forgot to update this

Comment (WyriHaximus) [reply README.md]: Hah yes, updated it to link to PHP's website

**Comment by admiralAwkbar** [comment]
@WyriHaximus the goal is to eventually break the linter to allow it to run these in parallel as well. Hopefully that helps

**Comment by WyriHaximus** [comment]
@admiralAwkbar that's awesome, and it would speed it up for PHP on bigger PR's. But let us cross that bridge when we get to it :).

**Comment by WyriHaximus** [comment]
@admiralAwkbar also update the PR just now as I missed adding PHP to `docs/disabling-linters.md`.

**Comment by admiralAwkbar** [comment]
Pulling into to branch for full CI

**Pull request closed without merging**